src/io.jl

```diff
@@ -223,10 +221,13 @@ function prepare_gen_data(
     keys = ["coords", "volt", "tech", "cap", "pfactor", "minload", "scap",
             "wcap", "shut2loadtime", "status"]
     for p in plants
-        push!(fplants, [Dict(k => v for (k,v) in zip(keys,g)) for g in p])
-        for g in fplants[end]
-            format_gen!(g)
+        plant_dict = [Dict(k => v for (k,v) in zip(keys,g)) for g in p]
+
+        for gen in plant_dict
+            format_gen!(gen)
         end
+
+        push!(fplants, plant_dict)
```

Comment:
```suggestion
        push!(fplants, collect(plant_dict))
```

Reply:
Each `fplant` is one Dict. We want a vector of dicts, not one large dict.

Reply:
yeah, sorry, this is only an acceptable change if we change it to a Generator above (i.e. do both or do neither) https://github.com/invenia/SyntheticGrids.jl/pull/17#discussion_r313322688

Reply:
we know need this for the behaviour to be correct :)